pyanaconda/modules/storage/bootloader/installation.py

```diff
@@ -142,6 +142,34 @@ def run(self):
             raise BootloaderInstallationError(str(e)) from None
 
 
+class InstallBootloaderTaskViaBootupd(Task):
+    """Installation task for the bootloader via bootupd"""
+
+    def __init__(self, storage, sysroot):
+        """Create a new task."""
+        super().__init__()
+        self._storage = storage
+        self._sysroot = sysroot
+
+    @property
+    def name(self):
+        return "Install the bootloader"
+
+    def run(self):
+        """Run the task.
+
+        :raise: BootloaderInstallationError if the installation fails
+        """
+        rc = execWithRedirect(
+            "bootupctl",
+            ["backend", "install", "--auto", "--with-static-configs",
```

Comment:
It's worth noting another *big* difference here vs how Anaconda works by default is that we use a *static* grub config.  This means we don't run `grub2-mkconfig` for every kernel update - which in turns we don't run `os-prober`.  And IMO `os-prober` is one of the worst chunks of code we ship that runs as root, scanning all your block devices to see if there's e.g. a BeOS or Windows partition there every time you just want to update your kernel...